client/src/templates/Challenges/classic/mobile-layout.tsx

```diff
@@ -45,6 +45,69 @@ class MobileLayout extends Component<MobileLayoutProps, MobileLayoutState> {
     });
   };
 
+  avoidWindowScrolling = (event: Event) => {
+    event.preventDefault();
+  };
+
+  setToolPanelPosition = () => {
+    const toolPanelGroup = (
+      document.getElementsByClassName(
+        'tool-panel-group-mobile'
+      ) as HTMLCollectionOf<HTMLElement>
+    )[0];
+
+    if (navigator.userAgent.match(/iPhone|Android.+Mobile/)) {
+      if (
+        visualViewport?.height &&
+        window.innerHeight > visualViewport.height
+      ) {
+        setTimeout(() => {
+          window.scrollTo(0, 0);
+
+          toolPanelGroup.style.position = 'absolute';
```

Comment:
The issue with position absolute is that it moves the styled element out of the Dom flow. In order to  not cover other elements, the mobile layout would need  button padding to adjust for the overlap.

Reply:
To simplify the logic, how about we absolute position the tool-panel initially using CSS and then if we detect changes if viewport, we adjust the `bottom` or `top` of the tool-panel

Reply:
I agree with your idea for simplifying the logic, so I will refactor the code to adjust the tool-panel buttons with only `bottom` and `top` property.